data/core/commandview.lua

```diff
@@ -274,18 +275,31 @@ function CommandView:get_suggestion_line_height()
 end
 
 
-function CommandView:update_suggestions()
-  local t = self.state.suggest(self:get_text()) or {}
+function CommandView:update_suggestions(preserve)
+  local text = self:get_text()
+  local t = self.state.suggest(text == self.save_suggestion and "" or text) or {}
   local res = {}
   for i, item in ipairs(t) do
     if type(item) == "string" then
       item = { text = item }
     end
     res[i] = item
   end
+  if preserve and self.suggestions and self.suggestion_idx ~= 1 then
+    local current_suggestion_idx
+    for i, v in ipairs(res) do
+      if v.text == self.suggestions[self.suggestion_idx].text then
+        current_suggestion_idx = i
+        break
+      end
+    end
+    self.suggestion_idx = current_suggestion_idx
+    self.suggestions_offset = 1
```

Comment:
Should we try to preserve the offset?
Does it currently move the "location" of the selection around?

Reply:
I think it's a pretty rare use case, and I'm not sure exactly how we would preserve it (though I haven't really thought much about it).

Reply:
In theory it should keep the current one, except when it violates the limits in `get_suggestions_offset`.

Reply:
OK, done. I think.